include/fiction/algorithms/path_finding/distance.hpp

```diff
@@ -71,23 +71,22 @@ template <typename Lyt, typename Dist = double>
  * @param c2 The second cell.
  * @return The distance between the two cells in nanometers.
  */
-template <typename Lyt, typename Dist = double>
-[[nodiscard]] constexpr Dist
+template <typename Lyt>
+[[nodiscard]] constexpr units::length::nanometer_t
 sidb_nanometer_distance([[maybe_unused]] const Lyt& lyt, const coordinate<Lyt>& source, const coordinate<Lyt>& target,
                         const sidb_simulation_parameters& sp = sidb_simulation_parameters{}) noexcept
 {
     static_assert(is_cell_level_layout_v<Lyt>, "Lyt is not a cell-level layout");
     static_assert(has_sidb_technology_v<Lyt>, "Lyt is not based on SiDB technology");
     static_assert(has_siqad_coord_v<Lyt>, "Lyt is not based on SiQAD coordinates");
-    static_assert(std::is_floating_point_v<Dist>, "Dist is not a floating-point type");
 
     const auto pos_c1 = sidb_nm_position<Lyt>(sp, source);
     const auto pos_c2 = sidb_nm_position<Lyt>(sp, target);
 
-    const auto x = static_cast<Dist>(pos_c1.first) - static_cast<double>(pos_c2.first);
-    const auto y = static_cast<Dist>(pos_c1.second) - static_cast<double>(pos_c2.second);
+    const auto x = pos_c1.first - pos_c2.first;
+    const auto y = pos_c1.second - pos_c2.second;
 
-    return static_cast<Dist>(std::hypot(x, y));
+    return {units::math::hypot(x, y)};
```

Comment:
Now with the usage of `units::math::hypot`, you probably don't need the braces anymore.